### PR TITLE
feat: make a basic device connection

### LINKED ISSUE #1250: Connect smart speaker with SUSI Android App
As discussed in the meeting, we need to add the feature to connect smart speakers with our Android app using Wifi-direct. 
@batbrain7 @imjog We can discuss here.

### LINKED ISSUE #1489: To follow the Smart Speaker- Android connection workflow similar to Google Home
**Actual Behaviour**

No connection workflow right now

**Expected Behaviour**

To follow a workflow similar to Google Home

**Would you like to work on the issue?**

### PR DESCRIPTION
**Pull request by batbrain7**
Fixes #1250 

This PR should not be reviewed on grounds for string resources and grammatical names as the strings will be deleted anyways, only the simplest connection possible is the aim of this PR.

A basic connection is established through the android app to the smart speaker. The API endpoints are hitting correctly, but due to recent code changes in the susi linux repository the code crashes on raspi, 
A follow up fixing PR has been sent for that by @stealthanthrax : https://github.com/fossasia/susi_linux/pull/258

here is the output  : 
8771327e-fd02-4808-820c-e23c8c11749f

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -252,6 +252,8 @@
 
     <!-- Devices String-->
     <string name="connect_success">Device Connected Successfully !</string>
+    <string name="auth_success">Auth credentials sent Successfully !</string>
+    <string name="wifi_success">Wifi credentials sent Successfully !</string>
```
Comment (arundhati24) [inline app/src/main/res/values/strings.xml]: Remove space before '!'

Comment (arundhati24) [reply app/src/main/res/values/strings.xml]: 'successfully' (small s)

File: app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/adapters/viewholders/WifiViewHolder.java [hunk 2]
```diff
@@ -19,6 +21,8 @@
 
     public @BindView(R.id.wifi_name)
     TextView wifiName;
+    public @BindView(R.id.layout_wifi)
+    LinearLayout layout;
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/adapters/viewholders/WifiViewHolder.java]: access modifier?

File: app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectPresenter.kt [hunk 3]
```diff
@@ -176,14 +181,21 @@ class DeviceConnectPresenter(deviceActivity: DeviceActivity, manager: WifiManage
...
     override fun makeConfigRequest() {
+        Timber.d("In here : CONFIG REQUEST")
+        deviceConnectView?.showProgress(utilModel.getString(R.string.connecting_device))
+        deviceModel.setConfiguration("google", "google", "y", "n", this@DeviceConnectPresenter)
     }
 
     override fun makeAuthRequest() {
+        Timber.d("In here : AUTH REQUEST")
+        deviceConnectView?.showProgress(utilModel.getString(R.string.connecting_device))
+        deviceModel.sendAuthCredentials("y", "mohitkumar2k15@dtu.ac.in", "batbrain", this@DeviceConnectPresenter)
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectPresenter.kt]: why hardcoded credentials?

File: app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectPresenter.kt [hunk 4]
```diff
@@ -143,31 +142,37 @@ class DeviceConnectPresenter(deviceActivity: DeviceActivity, manager: WifiManage
...
-        deviceConnectView?.onDeviceConnectionSuccess()
+        deviceConnectView?.onDeviceConnectionSuccess(utilModel.getString(R.string.wifi_success))
+        makeAuthRequest()
     }
 
     override fun onSendCredentialFailure() {
         Timber.d("WIFI - FAILURE")
-        //   deviceConnectView?.onDeviceConnectionError("Wifi Cred Failure", "Not done properly")
+        deviceConnectView?.stopProgress()
+        deviceConnectView?.onDeviceConnectionError("Wifi Cred Failure", "Not done properly")
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectPresenter.kt]: Cred ---> credentials, small f in Failure

File: app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectFragment.kt [hunk 5]
```diff
@@ -197,7 +206,7 @@ class DeviceConnectFragment : Fragment(), IDeviceConnectView {
             if (p1 != null) {
                 if (p1.action.equals(WifiManager.SCAN_RESULTS_AVAILABLE_ACTION)) {
                     val wifiList = mainWifi.getScanResults()
-                    Timber.d("Check%s",checkDevice)
+                    Timber.d("Check%s", checkDevice)
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectFragment.kt]: Not much imp but space can be added after Check i.e. "Check %s"

File: app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectPresenter.kt (hunk 4 above)
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectPresenter.kt]: Why hardcoded strings?

File: app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectFragment.kt [hunk 6]
```diff
@@ -12,6 +12,7 @@ import android.net.wifi.SupplicantState
 import android.net.wifi.WifiManager
 import android.os.Build
 import android.os.Bundle
+import android.os.Message
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectFragment.kt]: Where is this being used? Is this extra import?

File: app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectPresenter.kt (hunk 3 above)
Comment (batbrain7) [reply app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectPresenter.kt]: The PR is basic connection and in PRs earlier I have mentioned the reason for them, refer those PRs

File: app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectPresenter.kt (hunk 4 above)
Comment (batbrain7) [reply app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectPresenter.kt]: This is a basic connection, : 
as mario said on gitter first the base should be done and then modified, these strings won't even exist afterwards but right now these are important just to symbolize some work when we connect to the device.

File: app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectPresenter.kt (hunk 3 above)
Comment (arundhati24) [reply app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectPresenter.kt]: Sorry, can't recall specifically which one. Please link me to the PRs.

File: app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectFragment.kt (hunk 5 above)
Comment (batbrain7) [reply app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectFragment.kt]: I used default android studio styling so I don't know if this matters, plus it is a debug string so not subjected for review

File: app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectPresenter.kt (hunk 4 above)
Comment (batbrain7) [reply app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectPresenter.kt]: same as above

File: app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectFragment.kt (hunk 6 above)
Comment (batbrain7) [reply app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectFragment.kt]: No this is not

File: app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectFragment.kt (hunk 5 above)
Comment (arundhati24) [reply app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectFragment.kt]: This is not default styling. It depends on whether you add space or not. I already said its not imp but if you feel like doing this you may while making other changes. Just mentioned. Afterall many people work on the project together, so there's nothing wrong in keeping the messages/strings proper.

File: app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/adapters/viewholders/WifiViewHolder.java (hunk 2 above)
Comment (batbrain7) [reply app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/adapters/viewholders/WifiViewHolder.java]: yes it is required

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (batbrain7) [reply app/src/main/res/values/strings.xml]: not important, to be deleted anyways put only for testing purpose

File: app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/adapters/viewholders/WifiViewHolder.java (hunk 2 above)
Comment (arundhati24) [reply app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/adapters/viewholders/WifiViewHolder.java]: Yes, so please add it.

**Comment by batbrain7** [comment]
@arundhati24 refer the PR description and review

**Comment by arundhati24** [comment]
Okay @batbrain7 , when criteria for review have already been defined, I am approving.

**Comment by batbrain7** [comment]
@arundhati24 thanks :)

**Comment by arundhati24** [comment]
@batbrain7 I know things are high priority and I very well know we need to get PRs merged for continuing work, but I think this is not a personal project and we should be a bit patient. I think merging PRs on our own simply because we have write access is not at all justified. And, you could have waited for a review from @iamareebjamal 
I don't think this is a way to write just anything in the strings or debug messages and specify the criteria of review. It's an open source project and there are others also working on this project. And, there should be certain standards. I understand there is a need to implement basics first but that doesn't mean we can just write whatever we want. There should be certain basic standards. You know that these things will change in further PRs but not any new contributor who might see the project for the first time.

**Comment by batbrain7** [comment]
Okay @arundhati24 I am reverting the PR , I am sorry for merging the PR.

**Comment by arundhati24** [comment]
I don't need you to revert it nor do I have any issue with it. But, please consider that this is not a personal project but an open source project.

**Comment by iamareebjamal** [comment]
Please don't play merge revert game. Revert is not a simple undo, it's 3 commits instead. You should have not merged, and if it was merged should have let it be merged until a mentor got to review the situation

**Comment by iamareebjamal** [comment]
Please never merge your own PRs and wait for others to review, you would have seen that many PRs previously which were approved by as many as 7 members contained not only basic stylistic and logical issues, but bugs which later caused crashes in the app. The code quality and strictness in the project is going low recently, where all collaborators just approve the PR without checking the implementation and code design. MVP architecture is not being used and code is being added in activities, fragments. Remember that GSoC is about improving yourself to meet the standards of the project and then even improve it to new heights with ingenuity and better code, not to drag it down to your comfort level

**Comment by batbrain7** [comment]
I am really sorry @iamareebjamal @arundhati24 for merging the PR, I guess I was too focused on achieving the goal that I forgot the basic lessons of Open Source and contributing in a community , I completely take responsibility for my actions and promise that anything like this will never happen again.
Again I am really sorry

**Comment by batbrain7** [comment]
Also I don't mean any disrespect to maintainers and other collaborators, I just wanted to make this project better and achieve the goal what we set for GSoC and I really want this to be a learning experience and I have learnt a lot from the code reviews, this was a severe mistake from my side and I immediately regretted doing that.
 I am really sorry @iamareebjamal @arundhati24 .

**Pull request merged**